### PR TITLE
Add with_enumeration(enumeration) scope

### LINKED ISSUE #18: Add with_enumeration(enumeration) scope
Examples:

```
Post.with_status(:draft)
Post.with_status([:draft, :published])
```

### PR DESCRIPTION
**Pull request by domagojnakic**
- Add with_enumeration(enumeration) scope
- Add tests for with_enumeration(enumeration) scope
- Update readme

Resolves #18: Add with_enumeration(enumeration) scope.

### CODE REVIEW

File: lib/enumerations/value.rb [hunk 1]
```diff
@@ -16,6 +16,10 @@ def to_sym
       symbol
     end
 
+    def quoted_id
```
Comment (d4be4st) [inline lib/enumerations/value.rb]: Dont think this quoted id works as i initally thought. Remove this from this PR, and please add it to a separate issue like https://github.com/infinum/enumerations/issues/20

Comment (PetarCurkovic) [inline lib/enumerations/value.rb]: @d4be4st without `quoted_id`, with_enumeration(enumeration) scope will not work if we send Enumerations.value object:

`User.with_role(Role.admin, 'editor')`

Comment (d4be4st) [inline lib/enumerations/value.rb]: please be aware of this: https://github.com/rails/rails/pull/27962

**Comment by DamirSvrtan** [comment]
@d4be4st @PetarCurkovic - @cilim told me that we need the `qouted_id` functionality on our project from this PR, what's the status of this?

**Comment by PetarCurkovic** [comment]
@DamirSvrtan there is still a bug with `quoted_id` and `id` methods when you need to use enumerations in `where` clause. I'll try to fix it soon.

**Comment by DamirSvrtan** [comment]
@PetarCurkovic any progress with this?

**Pull request closed without merging**